### PR TITLE
Wire in Stake motion saga

### PR DESCRIPTION
**Pull request by willm30**
## Description

This PR wires in the stake motion saga. Test in conjunction with `block-ingestor`: [add/motion-staking-support](https://github.com/JoinColony/block-ingestor/pull/36)

It does not handle the response in the ui to motion staking; it just ensures the action is successfully performed on-chain.

## Testing

**To save yourself time**, I recommend testing this pr first: https://github.com/JoinColony/colonyCDapp/pull/338
Then come back here and just do code review.

**Changes** 🏗

* Adapted and wired in stakeMotion saga
* Passed motion id down to staking input

Contributes to #298

### CODE REVIEW

File: src/components/common/ColonyActions/ActionDetailsPage/DefaultMotion/MotionPhaseWidget/StakingWidget/StakingInput/StakingInput.tsx [hunk 1]
```diff
@@ -19,22 +22,45 @@ const validationSchema = object({
...
   //     setFieldValue('amount', 0);
   //     scrollToRef?.current?.scrollIntoView({ behavior: 'smooth' });
   //   },
 
+  const { user } = useAppContext();
+  const { colony } = useColonyContext();
+  const isObjection = false;
+  const vote = isObjection ? 0 : 1;
+
+  const transform = mapPayload(({ amount: sliderAmount }) => {
```
Comment (willm30) [inline src/components/common/ColonyActions/ActionDetailsPage/DefaultMotion/MotionPhaseWidget/StakingWidget/StakingInput/StakingInput.tsx:43]: fake payload for staking action

File: src/redux/sagas/transactions/createTransaction.ts [hunk 2]
```diff
@@ -131,3 +132,18 @@ export function* waitForTxResult(channel: Channel<any>) {
     throw new Error('Transaction failed');
   }
 }
+
+export const createGroupTransaction = (
```
Comment (willm30) [inline src/redux/sagas/transactions/createTransaction.ts:136]: #334

File: src/components/common/ColonyActions/ActionDetailsPage/DefaultMotion/MotionPhaseWidget/StakingWidget/StakingInput/StakingInput.tsx [hunk 3]
```diff
@@ -19,22 +22,45 @@ const validationSchema = object({
...
+    const finalStake = BigNumber.from('1000000000000000000').div(sliderAmount);
+    /* getFinalStakeFromSliderAmount(amount); */
+
+    return {
+      amount: finalStake,
+      userAddress: user?.walletAddress,
+      colonyAddress: colony?.colonyAddress,
+      motionId: BigNumber.from(motionId),
+      vote,
+    } as StakeMotionPayload;
```
Comment (jakubcolony) [inline src/components/common/ColonyActions/ActionDetailsPage/DefaultMotion/MotionPhaseWidget/StakingWidget/StakingInput/StakingInput.tsx:53]: Good idea!

**Review by jakubcolony — APPROVED** [review]
All looking good here 👍

File: amplify/backend/api/colonycdapp/schema.graphql [hunk 4]
```diff
@@ -441,6 +445,7 @@ type ColonyAction @model {
   type: ColonyActionType!
   blockNumber: Int!
   isMotion: Boolean
+  motionData: MotionData
```
Comment (rdig) [inline amplify/backend/api/colonycdapp/schema.graphql:448]: Why does need to be a separate type? Are we expecting it grow in field size?

File: src/components/common/ColonyActions/ActionDetailsPage/ActionDetailsPage.tsx [hunk 5]
```diff
@@ -45,8 +45,12 @@ const ActionDetailsPage = () => {
 
   const isMotion = action?.isMotion;
   const createdAt = action?.createdAt;
+  const isInvalidMotion = isMotion && !action.motionData;
   const isInvalidTransaction =
-    isInvalidTransactionHash || isUnknownTransaction || !action;
+    isInvalidTransactionHash ||
+    isUnknownTransaction ||
+    !action ||
+    isInvalidMotion;
```
Comment (rdig) [inline src/components/common/ColonyActions/ActionDetailsPage/ActionDetailsPage.tsx:53]: Most of these checks (at least the tx) one is a legacy, simple, check from back in the dapp days. It's basically a static string format check.

I'm not saying now, but in the future, this is good place we can refactor in order to have more smart checks (we can even check them via a block ingestor directly on chain, and keep a record of them, with various metadata, in the database)

File: src/components/common/ColonyActions/ActionDetailsPage/DefaultMotion/MotionPhaseWidget/StakingWidget/StakingInput/StakingInput.tsx [hunk 6]
```diff
@@ -19,22 +22,45 @@ const validationSchema = object({
...
   // const handleSuccess = (_, { setFieldValue, resetForm }) => {
   //     resetForm({});
   //     setFieldValue('amount', 0);
   //     scrollToRef?.current?.scrollIntoView({ behavior: 'smooth' });
   //   },
 
+  const { user } = useAppContext();
+  const { colony } = useColonyContext();
+  const isObjection = false;
+  const vote = isObjection ? 0 : 1;
```
Comment (rdig) [inline src/components/common/ColonyActions/ActionDetailsPage/DefaultMotion/MotionPhaseWidget/StakingWidget/StakingInput/StakingInput.tsx:41]: This should be a type (yay / nay), same way as on the block ingestor side

**Review by rdig — APPROVED** [review]
Left some comments, but otherwise I'm all good with this.

File: amplify/backend/api/colonycdapp/schema.graphql (hunk 4 above)
Comment (willm30) [reply amplify/backend/api/colonycdapp/schema.graphql:448]: Yes it gets bigger

File: src/components/common/ColonyActions/ActionDetailsPage/ActionDetailsPage.tsx (hunk 5 above)
Comment (willm30) [reply src/components/common/ColonyActions/ActionDetailsPage/ActionDetailsPage.tsx:53]: https://github.com/JoinColony/colonyCDapp/issues/347

**Pull request merged**